src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/PooledObjects/PooledArrayBuilder`1.cs

```diff
@@ -100,4 +133,33 @@ public void ClearAndFree()
 
     public readonly T[] ToArray()
         => _builder?.ToArray() ?? Array.Empty<T>();
+
+    public void Push(T item)
```

Comment:
I did try to make these extension methods, but you can't call an extension method that takes `this` by `ref` on a `using` variable.

Reply:
Works for me. Honestly, I don't think there's much value in making this an extension method anyway.